control-plane/stor-port/src/types/v0/store/snapshots/replica.rs

```diff
@@ -0,0 +1,52 @@
+use super::SnapshotSpec;
+use crate::types::v0::{
+    store::{OperationSequence, SpecStatus},
+    transport::ReplicaId,
+};
+
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+/// User specification of a replica snapshot.
+pub type ReplicaSnapshotSpec = SnapshotSpec<ReplicaId>;
+/// State of the ReplicaSnapshotSpec Spec.
+pub type ReplicaSnapshotSpecStatus = SpecStatus<()>;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default)]
+pub struct ReplicaSnapshot {
+    /// Status of the replica snapshot.
+    status: ReplicaSnapshotSpecStatus,
+    /// User specification of the snapshot.
+    spec: ReplicaSnapshotSpec,
+    /// Control-plane related information of the snapshot (book-keeping).
+    metadata: ReplicaSnapshotMeta,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default)]
+pub struct ReplicaSnapshotMeta {
+    #[serde(skip)]
+    sequencer: OperationSequence,
+    /// Record of the operation in progress.
+    operation: Option<ReplicaSnapshotOperationState>,
+
+    /// Creation timestamp of the snapshot (set after creation time).
+    creation_timestamp: Option<DateTime<Utc>>,
```

Comment:
How should we store timestamp in etcd, epoch based counter or using formatted strings like DateTime ?

Reply:
I don't think there would be a difference in terms of the how much space it takes to store in whatever way. Considering that, human readable format can be preferred.